filesystem/src/main/java/org/frankframework/filesystem/AbstractFileSystemListener.java

```diff
@@ -400,25 +415,49 @@ public RawMessageWrapper<F> changeProcessState(RawMessageWrapper<F> message, Pro
 			}
 			if (toState==ProcessState.INPROCESS && isFileTimeSensitive() && getFileSystem() instanceof IWritableFileSystem) {
 				F movedFile = getFileSystem().moveFile(message.getRawMessage(), getStateFolder(toState), false);
-				String newName = getFileSystem().getCanonicalName(movedFile)+"-"+(DateFormatUtils.format(getFileSystem().getModificationTime(movedFile), DateFormatUtils.FULL_ISO_TIMESTAMP_NO_TZ_FORMATTER).replace(":", "_"));
-				F renamedFile = getFileSystem().toFile(newName);
-				int i=1;
-				while(getFileSystem().exists(renamedFile)) {
-					renamedFile=getFileSystem().toFile(newName+"-"+i);
-					if(i>5) {
-						log.warn("Cannot rename file [{}] with the timestamp suffix. File moved to [{}] folder with the original name", ()->message, ()->getStateFolder(toState));
-						return wrap(movedFile, message);
-					}
-					i++;
-				}
-				return wrap(FileSystemUtils.renameFile((IWritableFileSystem<F>) getFileSystem(), movedFile, renamedFile, false, 0), message);
+				return wrap(renameFileWithTimeStamp(message, toState, movedFile), message);
```

Comment:
Extracted this logic to keep code readable while adding more functionality to it.